test/operatortest.jl

```diff
@@ -0,0 +1,8 @@
+#######################
+# Spin Operators Test #
+#######################
+
+@assert commutator(sigma_x, sigma_y) == 2*im*sigma_z
```

Comment:
This is really nice! Could you add a similar test for position and momentum operators (just use some reasonable basis size).